istioctl/cmd/version.go

```diff
@@ -98,10 +99,14 @@ func getProxyInfo() (*[]istioVersion.ProxyInfo, error) {
 		}
 
 		for _, ss := range sss {
-			pi = append(pi, istioVersion.ProxyInfo{
-				ID:           ss.ProxyID,
-				IstioVersion: ss.SyncStatus.IstioVersion,
-			})
+			// if proxy id has istio gateway items, let us exclude them
+			// from data plane as they are also shown under control plane
+			if !strings.HasPrefix(ss.ProxyID, "istio-ingressgateway-") && !strings.HasPrefix(ss.ProxyID, "istio-egressgateway-") && !strings.HasPrefix(ss.ProxyID, "prometheus-") {
```

Comment:
istio-ingressgateway is not a special name, we just happen to name it this way. These are very much a part of the data plane. If anything, we should remove them from the control plane list in my opinion.

Reply:
I do agree the gateways are part of the data plane conceptually, however from the upgrade perspective, users are thinking they are part of Istio and updated as part of Istio control plane update per our doc:  https://istio.io/docs/setup/upgrade/istioctl-upgrade/#upgrade-steps. cc @kramvan1 @esnible for input

Reply:
To me, the istioctl version cmd for the data plane section is mostly used for evaluating if the data plane sidecars need to be updated.